src/bin/cratesfyi.rs

```diff
@@ -325,3 +36,402 @@ fn logger_init() {
+                let mut builder = RustwideBuilder::init().unwrap();
+                builder
+                    .add_essential_files()
+                    .expect("failed to add essential files");
+            }
+
+            Self::Lock => docbuilder.lock().expect("Failed to lock"),
+            Self::Unlock => docbuilder.unlock().expect("Failed to unlock"),
+            Self::PrintOptions => println!("{:?}", docbuilder.options()),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, StructOpt)]
+enum DatabaseSubcommand {
+    #[structopt(name = "move-to-s3")]
+    MoveToS3,
+
+    /// Run database migrations
+    Migrate {
+        /// The database version to migrate to
+        #[structopt(name = "VERSION")]
+        version: i64,
+    },
+
+    /// Updates github stats for crates.
+    UpdateGithubFields,
+
+    AddDirectory {
+        /// Path of file or directory
+        #[structopt(name = "DIRECTORY")]
+        directory: PathBuf,
+        /// Prefix of files in database
+        #[structopt(name = "PREFIX", env = "CRATESFYI_PREFIX")]
+        prefix: String,
+    },
+
+    /// Updates monthly release activity chart
+    UpdateReleaseActivity,
+
+    /// Updates search index
+    UpdateSearchIndex,
+
+    /// Removes a whole crate from the database
+    DeleteCrate {
+        /// Name of the crate to delete
+        #[structopt(name = "CRATE_NAME")]
+        crate_name: String,
+    },
+
+    /// Blacklist operations
+    Blacklist {
+        #[structopt(subcommand)]
+        command: BlacklistSubcommand,
+    },
+}
+
+impl DatabaseSubcommand {
+    pub fn handle_args(self) {
+        match self {
+            Self::MoveToS3 => {
+                let conn = db::connect_db().expect("failed to connect to the database");
+
+                let (mut count, mut total) = (1, 0);
+                while count != 0 {
+                    count = db::move_to_s3(&conn, 5_000).expect("Failed to upload batch to S3");
+                    total += count;
+
+                    eprintln!(
+                        "moved {} rows to s3 in this batch, total moved so far: {}",
+                        count, total
+                    );
+                }
+            }
+
+            Self::Migrate { version } => {
+                let conn = connect_db().expect("failed to connect to the database");
+                db::migrate(Some(version), &conn).expect("Failed to run database migrations");
+            }
+
+            Self::UpdateGithubFields => {
+                cratesfyi::utils::github_updater().expect("Failed to update github fields");
+            }
+
+            Self::AddDirectory { directory, prefix } => {
+                let conn = db::connect_db().expect("failed to connect to the database");
+                add_path_into_database(&conn, &prefix, directory)
+                    .expect("Failed to add directory into database");
+            }
+
+            // FIXME: This is actually util command not database
```

Comment:
`util` is mostly database operations and doesn't have it's own subcommand.

Reply:
The fixme wasn’t mine, I just kept it